### PR TITLE
Update to Ruby 3.2 and latest Jekyll

### LINKED ISSUE #347: Failed build when running locally
## Describe the bug

I'm unable to get the project up and running due to an error when following the instructions as-is.

## To Reproduce

1. Open the Terminal
2. Execute:
```bash
git clone https://github.com/apple/swift-org-website.git
cd swift-org-website
bundle install
LC_ALL=en_us.UTF-8 bundle exec jekyll serve
open "http://localhost:4000"
```
3. See error

## Expected behavior

The project is up and running and served via `http://localhost:4000`

## Actual behavior

The following error is thrown:

```bash
# ...
Using terminal-table 2.0.0
Using jekyll 4.2.2
Using jekyll-redirect-from 0.16.0
Using webrick 1.7.0
Bundle complete! 3 Gemfile dependencies, 30 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Configuration file: /Users/<username>/programming/swift/playground/swift-org-website/_config.yml
            Source: /Users/<username>/programming/swift/playground/swift-org-website
       Destination: /Users/<username>/programming/swift/playground/swift-org-website/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
Liquid Exception: undefined method `tainted?' for "The Swift.org Blog":String in /Users/<username>/programming/swift/playground/swift-org-website/_layouts/post.html
                    ------------------------------------------------
      Jekyll 4.2.2   Please append `--trace` to the `serve` command
                     for any additional information or backtrace.
                    ------------------------------------------------
/opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/variable.rb:124:in `taint_check': undefined method `tainted?' for "The Swift.org Blog":String (NoMethodError)

      return unless obj.tainted?
                       ^^^^^^^^^
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/variable.rb:89:in `render'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:103:in `render_node_to_output'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:80:in `render'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/template.rb:208:in `block in render'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/template.rb:242:in `with_profiling'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/template.rb:207:in `render'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/liquid-4.0.3/lib/liquid/template.rb:220:in `render!'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:39:in `block (3 levels) in render!'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:59:in `measure_counts'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:38:in `block (2 levels) in render!'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:63:in `measure_bytes'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:37:in `block in render!'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:70:in `measure_time'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/liquid_renderer/file.rb:36:in `render!'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/renderer.rb:131:in `render_liquid'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/renderer.rb:194:in `render_layout'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/renderer.rb:163:in `place_in_layouts'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/renderer.rb:93:in `render_document'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/renderer.rb:63:in `run'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:547:in `render_regenerated'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:532:in `block (2 levels) in render_docs'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:531:in `each'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:531:in `block in render_docs'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:530:in `each_value'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:530:in `render_docs'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:210:in `render'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/site.rb:80:in `process'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/command.rb:28:in `process_site'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/commands/build.rb:65:in `build'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/commands/build.rb:36:in `process'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `each'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/jekyll-4.2.2/exe/jekyll:15:in `<top (required)>'
	from /opt/homebrew/lib/ruby/gems/3.2.0/bin/jekyll:25:in `load'
	from /opt/homebrew/lib/ruby/gems/3.2.0/bin/jekyll:25:in `<top (required)>'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:58:in `load'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:23:in `run'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli.rb:483:in `exec'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli.rb:31:in `dispatch'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/cli.rb:25:in `start'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/exe/bundle:48:in `block in <top (required)>'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /opt/homebrew/lib/ruby/gems/3.2.0/gems/bundler-2.3.13/exe/bundle:36:in `<top (required)>'
	from /opt/homebrew/opt/ruby/bin/bundle:25:in `load'
	from /opt/homebrew/opt/ruby/bin/bundle:25:in `<main>'
```

## Screenshots

Screenshot 2023-07-24 at 8 27 21 PM

### OS (please complete the following information):

- macOS
- Version: 13.4 (22F66)
- Apple Swift version 5.8.1 (swiftlang-5.8.0.124.5 clang-1403.0.22.11.100)
- Target: arm64-apple-macosx13.0
- ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]

### Swift.org version/commit hash

commit 8ed8ac5ad1fb1b36fcf6912d5783fce87433342f

### PR DESCRIPTION
**Pull request by daveverwer**
Fixes #347

As Tim [suggested here](https://github.com/apple/swift-org-website/issues/347#issuecomment-1656218454), this fixes the site build process when trying to use Ruby 3.2 to build it.

- [x] Updated recommended version of Ruby to the latest stable, 3.2.0.
- [x] Removed unnecessary Ruby version comment from the top of the `Gemfile`.
- [x] Bundle update, including updating Jekyll to the latest 4.3.3.
- [x] `Dockerfile` updated to use Ruby 3.2 base image and latest Bundler version.
- [x] Sass files updated to parse correctly with the latest Jekyll.

### CODE REVIEW

**Comment by daveverwer** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
lgtm

**Review by tomerd — APPROVED** [review]

**Review by 0xTim — APPROVED** [review]

**Review by alexandersandberg — APPROVED** [review]

**Comment by daveverwer** [comment]
@shahmishal Are you happy with this? OK to merge?

**Comment by daveverwer** [comment]
@swift-ci please test

**Review by kaishin — APPROVED** [review]

**Review by shahmishal — APPROVED** [review]

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by daveverwer** [comment]
@swift-ci please test

**Comment by daveverwer** [comment]
@shahmishal this came up again just now in #475. We should try and get this merged if possible.

I just rebased it and did another `bundle update` to integrate gem changes since it was opened so it's still ready to go from my side.

**Comment by daveverwer** [comment]
Rebased once more for discussion at tonight's SWWG meeting.

**Comment by daveverwer** [comment]
@swift-ci please test

**Comment by daveverwer** [comment]
@swift-ci please test

**Comment by daveverwer** [comment]
@swift-ci please test

File: assets/stylesheets/_mobile.scss [hunk 1]
```diff
@@ -85,7 +85,7 @@
     }
 }
 
-@media only screen and(max-width: 767px) {
+@media only screen and (max-width: 767px) {
```
Comment (daveverwer) [inline assets/stylesheets/_mobile.scss:88]: This whitespace is now required by the latest sass parser included with the latest Jekyll. See below for more instances of this.

**Comment by daveverwer** [comment]
@swift-ci please test

**Comment by daveverwer** [comment]
@shahmishal over to you!

**Pull request merged**